### ISSUE TITLE
LinuxNetworkManager is now available for Open, WPA, WPA2 and WEP wireless

### ISSUE THREAD

**Issue by WesGilster** [area515/Creation-Workshop-Host#74]
I suppose it seems counter-intuitive to have a web based network manager, but this feature coupled with the multicast printer detection will eliminate all setup from an end user standpoint assuming an initial wired Ethernet connection. No more fiddling with the wpa_supplicant. It will scan the wireless networks in range, you'll select a network and connect with the password . You won't even have to select the wireless encryption type.

The following command is a cli example implementation(in the test kit) of what is available via the restful interface:
sudo java -cp lib/*:. org.area515.resinprinter.test.LinuxManualWifiConnect

This has been tested on WEP networks.

Version cwhTestKit-0.156

**Comment by WesGilster** [comment]
This has now been tested on WPA networks.

**Comment by WesGilster** [comment]
This has now been tested on WPA/WPA2 transition networks with an emphasis on WPA2.

**Comment by WesGilster** [comment]
I still need to detect connection failures and report back to the client so that they don't believe Wifi setup was a success when it could have failed.

Wifi setup and email setup is now available from the new GUI only.

**Comment by WesGilster** [comment]
I'm probably going to change the way the host responds to a wifi reconfiguration by stop restarting CWH. I figured we would need a host restart after setting up Wifi and unplugging the Ethernet cable. Instead I found that Java doesn't skip a beat and the server socket keeps on trucking. I assume it's because Jetty is binding to 0.0.0.0 instead of a specific interface. I've even tested after performing invalid setups, after performing incorrect passphrases, using dynamic IPs and reconfiguration across different subnets. Still, a bit more testing is required.

I have noticed that the Raspberry Pi likes to stay in roaming mode, I could change this, but I'm assuming that the community likes this option. I find it very annoying since the ifplugd daemon kills wifi on startup if Ethernet is available and or if it ever gets plugged in.
